src/crypto/PSAOperationalKeystore.cpp

```diff
@@ -209,5 +266,32 @@ void PSAOperationalKeystore::ReleasePendingKeypair()
     mIsPendingKeypairActive = false;
 }
 
+CHIP_ERROR PSAOperationalKeystore::MigrateOpKeypairForFabric(FabricIndex fabricIndex,
+                                                             OperationalKeystore & operationalKeystore) const
+{
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+    VerifyOrReturnError(fabricIndex != kUndefinedFabricIndex, CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    P256SerializedKeypair serializedKeypair;
+
+    CHIP_ERROR err = operationalKeystore.ExportOpKeypairForFabric(fabricIndex, serializedKeypair);
+
+    if (CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND != err)
+    {
+        ReturnErrorOnFailure(err);
+
+        // Do not allow overwriting the existing key and just remove it from KVS
+        if (!HasOpKeypairForFabric(fabricIndex))
+        {
+            PersistentP256Keypair keypair(fabricIndex);
+            ReturnErrorOnFailure(keypair.Deserialize(serializedKeypair));
+        }
+
+        ReturnErrorOnFailure(operationalKeystore.RemoveOpKeypairForFabric(fabricIndex));
+    }
+
+    return err;
```

Comment:
You seem to meant ignoring CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND error but you still return it here?